modules/axtask/src/wait_queue.rs

```diff
@@ -148,6 +147,7 @@ impl WaitQueue {
 
         let mut timeout = true;
         while axhal::time::wall_time() < deadline {
+            let mut rq = current_run_queue::<NoPreemptIrqSave>();
```

Comment:
What about the timer event triggering between `while ...` and `let mut rq = ...`? Then the task will block and the timeout will not take effect.

Reply:
you are right, it has been fixed.

Reply:
A better solution is to still keep the guard in the loop, but also move the deadline check in the guard: 

```rs
loop {
    let mut rq = current_run_queue::<NoPreemptIrqSave>();
    if axhal::time::wall_time() >= deadline {
        break;
    }
}
```

Reply:
Okay, it has been modified to 
```Rust
        loop {
            let mut rq = current_run_queue::<NoPreemptIrqSave>();
            if axhal::time::wall_time() >= deadline {
                break;
            }
            let wq = self.queue.lock();
            if condition() {
                timeout = false;
                break;
            }

            rq.blocked_resched(wq);
        }
```